compiler/rustc_mir/src/const_eval/error.rs

```diff
@@ -150,28 +159,21 @@ impl<'tcx> ConstEvalErr<'tcx> {
         emit: impl FnOnce(DiagnosticBuilder<'_>),
         lint_root: Option<hir::HirId>,
     ) -> ErrorHandled {
-        let must_error = match self.error {
+        let must_error = match &self.error {
             err_inval!(Layout(LayoutError::Unknown(_))) | err_inval!(TooGeneric) => {
                 return ErrorHandled::TooGeneric;
             }
             err_inval!(AlreadyReported(error_reported)) => {
-                return ErrorHandled::Reported(error_reported);
+                return ErrorHandled::Reported(*error_reported);
             }
             // We must *always* hard error on these, even if the caller wants just a lint.
             err_inval!(Layout(LayoutError::SizeOverflow(_))) => true,
+            InterpError::MachineStop(err) => err.is_hard_err(),
```

Comment:
I don't think this is right. Check the comment below where `must_error` is used:
```rust
        if must_error {
            // The `message` makes little sense here, this is a more serious error than the
            // caller thinks anyway.
            // See <https://github.com/rust-lang/rust/pull/63152>.
```
We are now entering that code path for panics, aren't we?

Reply:
I guess ignoring the `message` works out in this case... but the comment is still wrong or at least misleading now.